lib/lambda_ethereum_consensus/p2p/incoming_requests_handler.ex

```diff
@@ -1,35 +1,55 @@
-defmodule LambdaEthereumConsensus.P2P.IncomingRequests.Handler do
+defmodule LambdaEthereumConsensus.P2P.IncomingRequestsHandler do
   @moduledoc """
   This module handles Req/Resp domain requests.
   """
-  require Logger
 
   alias LambdaEthereumConsensus.ForkChoice
-  alias LambdaEthereumConsensus.Libp2pPort
   alias LambdaEthereumConsensus.P2P.Metadata
   alias LambdaEthereumConsensus.P2P.ReqResp
   alias LambdaEthereumConsensus.Store.BlockDb
   alias LambdaEthereumConsensus.Store.Blocks
 
   require Logger
 
-  @spec handle(String.t(), String.t(), binary()) :: any()
-  def handle(name, message_id, message) do
-    case handle_req(name, message_id, message) do
-      :ok -> :ok
-      {:error, error} -> Logger.error("[#{name}] Request error: #{inspect(error)}")
+  @request_prefix "/eth2/beacon_chain/req/"
+  @request_names [
+    "status/1",
+    "goodbye/1",
+    "ping/1",
+    "beacon_blocks_by_range/2",
+    "beacon_blocks_by_root/2",
+    "metadata/2"
+  ]
+
+  @spec protocol_ids() :: list(String.t())
+  def protocol_ids() do
+    @request_names |> Enum.map(&Enum.join([@request_prefix, &1, "/ssz_snappy"]))
+  end
+
+  @spec handle(String.t(), String.t(), binary()) :: :ok | {:ok, any()} | {:error, String.t()}
+  def handle(@request_prefix <> name, message_id, message) do
+    Logger.debug("'#{name}' request received")
+
+    result =
+      :telemetry.span([:port, :request], %{}, fn ->
+        {handle_req(name, message_id, message), %{module: "handler", request: inspect(name)}}
+      end)
+
+    case result do
+      {:error, error} -> {:error, "[#{name}] Request error: #{inspect(error)}"}
+      result -> result
     end
   end
 
   @spec handle_req(String.t(), String.t(), binary()) ::
-          :ok | {:error, String.t()}
+          :ok | {:ok, any()} | {:error, String.t()}
```

Comment:
I'm not sure it's possible to return just `:ok` here? You either return an `{:ok, {message_id, payload}}` tuple that you wrote manually in each one or the with fails, I assume with an error tuple. I'd remove that from the spec.